### PR TITLE
Support mutiple remote analyzers

### PR DESCRIPTION
**Pull request by La0**
Nedds to merge #1953 first, then rebase

### CODE REVIEW

**Comment by La0** [comment]
Awaiting landing of [clang-format JSON output](https://phabricator.services.mozilla.com/D24194)

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py [hunk 1]
```diff
@@ -122,8 +123,8 @@ def get_eligible_files():
             groups.append(group)
 
             issues += [
-                ClangFormatIssue(filename, group[0], len(group), revision)
-                for group in groups
+                ClangFormatIssue(filename, g[0], len(group), revision)
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/format.py]: Why the change from `group` to `g`?
I guess you forgot to change `len(group)` to `len(g)`?

File: src/staticanalysis/bot/static_analysis_bot/clang/tidy.py [hunk 2]
```diff
@@ -234,19 +244,20 @@ class ClangTidyIssue(Issue):
...
-        self.line = int(self.line)
+        self.line = int(line)
         self.nb_lines = 1  # Only 1 line affected on clang-tidy
-        self.char = int(self.char)
+        self.char = int(char)
+        self.check = check
+        self.message = message
         self.body = None
         self.notes = []
+        self.type = type
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/tidy.py]: Nit: maybe use a different variable name since `type` is a keyword

File: src/staticanalysis/bot/static_analysis_bot/task.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        if self.state not in self.valid_states:
+            logger.warn('Invalid task state', state=self.state, id=self.id, name=self.name)
+            return
+
+        # Load relevant artifacts
+        out = {}
+        for artifact_name in self.artifacts:
+            logger.info('Load artifact', task_id=self.id, artifact=artifact_name)
+            try:
+                artifact = queue_service.getArtifact(self.id, self.run_id, artifact_name)
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/task.py]: Does `getArtifact` automatically retry? We should retry a few times on failure, as they are uploaded on S3 and there could be temporary failures.

File: src/staticanalysis/bot/static_analysis_bot/workflows/remote.py [hunk 4]
```diff
@@ -115,47 +57,32 @@ def run(self, revision):
...
-                    issues += self.parse_issues(task, log, revision)
+                task = self.build_task(dep_id, tasks[dep_id])
+                artifacts = task.load_artifacts(self.queue_service)
+                if artifacts is not None:
+                    task_issues = task.parse_issues(artifacts, revision)
+                    logger.info('Found {} issues'.format(len(task_issues)), task=task.name, id=task.id)
+                    issues += task_issues
             except Exception as e:
                 logger.warn('Failure during task analysis', task=settings.taskcluster.task_id, error=e)
-                raise
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/workflows/remote.py]: Maybe we should keep raising, since the analysis is now very simple (just parsing some artifacts) we should never fail

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py (hunk 1 above)
Comment (La0) [reply src/staticanalysis/bot/static_analysis_bot/clang/format.py]: Because pyflakes was complaining about re-defining the `group` variable from above.

File: src/staticanalysis/bot/static_analysis_bot/clang/tidy.py (hunk 2 above)
Comment (La0) [reply src/staticanalysis/bot/static_analysis_bot/clang/tidy.py]: Updated the local usage, and created a [good-first-bug issue](https://github.com/mozilla/release-services/issues/1980) to do it everywhere

File: src/staticanalysis/bot/static_analysis_bot/task.py (hunk 3 above)
Comment (La0) [reply src/staticanalysis/bot/static_analysis_bot/task.py]: It does up to 5 retries [when a 50x error occurs](https://github.com/taskcluster/taskcluster-client.py/blob/5ae57f12bfb56e3b7a7a150f655b8e76cb68ccbd/taskcluster/queue.py#L531)

Here is the [low level code](https://github.com/taskcluster/taskcluster-client.py/blob/43b29b63b4776360c8faf667a0182891cceaf362/taskcluster/client.py#L458) doing that in taskcluster-client.py

File: src/staticanalysis/bot/static_analysis_bot/workflows/remote.py (hunk 4 above)
Comment (La0) [reply src/staticanalysis/bot/static_analysis_bot/workflows/remote.py]: :+1: Hardcore mode.
